### PR TITLE
feat: add gradle support and update project structure

### PR DESCRIPTION
**Pull request by yaohui-wyh**
This PR helps refactoring the IntelliJ plugin project by adding Gradle build system (which is the recommended way for IntelliJ plugin development: <https://plugins.jetbrains.com/docs/intellij/gradle-build-system.html>). 

None of the current plugin features are modified. However, after restructuring it could make future development & release process easier.

Details:
- use Gradle for build configuration instead of IDEA's misc.xml. Developers using IDEA module configuration setups (misc.xml) could just delete the `.idea` folder and reimport the project
- move java classes into `src/main/java` source set (for IDEA to reload & re-index the Gradle project)
- move resources into `src/main/resources`
- update .gitignore
- update development guide in README.md

Tests:
- To run IDE with plugin installed: `./gradlew runIde` (checkout available [gradle tasks](https://github.com/JetBrains/gradle-intellij-plugin#tasks) for details)
- To create plugin artifact: `./gradlew buildPlugin`

Resources:

- https://github.com/JetBrains/intellij-platform-plugin-template
- https://github.com/JetBrains/intellij-plugins

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,7 +62,9 @@ Please file an issue: https://github.com/sourcegraph/sourcegraph-jetbrains/issue
 ## Development
 
 - Start IntelliJ and choose `Check out from Version Control` -> `Git` -> `https://github.com/sourcegraph/sourcegraph-jetbrains`
-- Develop as you would normally (hit Debug icon in top right of IntelliJ).
+- Develop as you would normally (hit Debug icon in top right of IntelliJ) or using gradlew commands:
+  1. `./gradlew runIde` to run an IDE instance with sourcegraph plugin installed
+  2. `./gradlew buildPlugin` to build plugin artifact (`build/distributions/Sourcegraph.zip`)
```
Comment (Strum355) [inline README.md:67]: Im curious if the line [here](https://github.com/sourcegraph/sourcegraph-jetbrains/pull/18/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R72) needs to be updated for the introduction of this new task and/or what the difference is between the two? I'm unfamiliar with Intellij plugin development as a whole :slightly_smiling_face:

**Review by Strum355 — APPROVED** [review]
Thanks for the PR! This seems all fine, I've one question about the instructions in the README, but everything else seems to be okay from what I know of this process :+1:

**Review by olafurpg — CHANGES_REQUESTED** [review]
Thank you for this contribution! This looks great 👏 I was able to run `./gradlew runIde` locally and it started a new instance of IntellJ with the plugin installed.

Only one blocking issue: can you please run `gradle wrapper` and commit the generated files? The command failed for me on the first run with the following error message

```
❯ ./gradlew runIde
Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain
```

The changes should look like this commit https://github.com/sourcegraph/sourcegraph-jetbrains/commit/d971d192a7fa758a2516e058af014b01d16f03c6

**Review by olafurpg — APPROVED** [review]
LGTM 👍 Thank you!

**Comment by yaohui-wyh** [comment]
Thanks @olafurpg. Previously all jar files are excluded and I just re-run the gradle task and commit the generated files.

**Comment by olafurpg** [comment]
@yaohui-wyh Yeah, I just realized that the commit I linked in the review didn't even contain the jar file because it was gitignored 😅

File: README.md (hunk 1 above)
Comment (yaohui-wyh) [reply README.md:67]: Hi @Strum355! `gradle runIde` launch an IDE instance with the installed plugin, which is used for local dev/test. You could also set up a Run/Debug configuration for it.
image

`gradle buildPlugin` trigger the build process and generate a packaged jar/zip file which can be later published to the [IntelliJ plugin repository](https://plugins.jetbrains.com/) or installed to IDE by user from local disk. The plugin jar/zip file could be delivered to users for testing/dogfooding before released to plugin repository.

image

**Pull request merged**